File: includes/team.py
class Team():

    def __init__(self,name,next = None):
        self._name = name
        self._rating = 0.0
        self._performance = 0.0
        self._scheduleFactor = 0.0
        self._wins = 0
        self._losses = 0
        self._ties = 0
        self.next = next

    def getTeam(self):
        return self._name

    def getRating(self):
        return float(self._rating)

    def setRating(self, rating):
        self._rating = rating

    def getPerformance(self):
        return self._performance

    def setPerformance(self, performance):
        self._performance = performance

    def getScheduleFactor(self):
        return float(self._scheduleFactor)

    def setScheduleFactor(self, factor):
        self._scheduleFactor = factor

    def getWins(self):
        return int(self._wins)

    def setWins(self, wins):
        self._wins = wins

    def getLosses(self):
        return int(self._losses)

    def setLosses(self, losses):
        self._losses = losses

    def getTies(self):
        return int(self._ties)

    def setTies(self, ties):
        self._ties = ties

File: cpp/src/team.cpp
#include "team.h"

Team::Team(string name, string next)
{
    name_ = name;
    rating_ = 0.0;
    performance_ = 0.0;
    scheduleFactor_ = 0.0;
    wins_ = 0;
    losses_ = 0;
    ties_ = 0;
    next_ = next;
}

string Team::getName()
{
    return name_;
}

double Team::getRating()
{
    return rating_;
}

double Team::getPerformance()
{
    return performance_;
}

double Team::getScheduleFactor()
{
    return scheduleFactor_;
}

int Team::getWins()
{
    return wins_;
}

int Team::getLosses()
{
    return losses_;
}

int Team::getTies()
{
    return ties_;
}

void Team::setRating(double rating)
{
    rating_ = rating;
}

void Team::setPerformance(double performance)
{
    performance_ = performance;
}

void Team::setScheduleFactor(double factor)
{
    scheduleFactor_ = factor;
}

void Team::setWins(int wins)
{
    wins_ = wins;
}

void Team::setLosses(int losses)
{
    losses_ = losses;
}

void Team::setTies(int ties)
{
    ties_ = ties;
}

Node Team::next()
{
    return next_;
}
